src/scalib_ext/scalib/src/sasca/bp_compute.rs

```diff
@@ -218,6 +218,30 @@ impl Distribution {
         );
         return res;
     }
+
+    // Update belief with weight average of new and previous value
+    pub fn update_dampen(
+        &mut self,
+        numerator: &Distribution,
+        denominator: &Distribution,
+        alpha: f64,
+    ) {
+        let mut self_arr = self
+            .value_mut()
+            .expect("update_dampen needs full dist self");
+        let num_arr = numerator
+            .value()
+            .expect("update_dampen needs full dist num");
+        let v: f64 = self_arr.iter().sum::<f64>();
+        if let Some(denom_arr) = denominator.value() {
+            let u: f64 = num_arr.iter().sum::<f64>() / denom_arr.iter().sum::<f64>();
+            azip!((x in &mut self_arr, y in &num_arr, z in &denom_arr) *x = (alpha/u) * (y/(z + MIN_PROBA)) + ((1.0 - alpha)/v) * *x)
```

Comment:
The addition of `MIN_PROBA` should not be needed. (z is a product of beliefs from factors, which are properly regularized.)